### PR TITLE
AppStream metainfo fixes

### PR DESCRIPTION
**Pull request by Be-ing**

### CODE REVIEW

**Comment by uklotzde** [comment]
This change might break existing packaging recipes and requires a note for maintainers.

File: res/linux/org.mixxx.Mixxx.metainfo.xml
Comment (uklotzde) [inline res/linux/org.mixxx.Mixxx.metainfo.xml:10]: content_rating is optional. Why do we need to add it?

Comment (Be-ing) [reply res/linux/org.mixxx.Mixxx.metainfo.xml:10]: Flathub builds fail without it.

**Comment by Be-ing** [comment]
I'll add a note to packagers in the CHANGELOG after both this and #4204 are merged.

File: res/linux/org.mixxx.Mixxx.metainfo.xml
Comment (uklotzde) [inline res/linux/org.mixxx.Mixxx.metainfo.xml:11]: Is this correct? Release tags must be encapsulated within a `<releases>` element.

I am not keen on maintaining yet another literal version constant in the codebase. Do we have other options?

**Review by uklotzde — CHANGES_REQUESTED** [review]

File: res/linux/org.mixxx.Mixxx.metainfo.xml
Comment (Be-ing) [reply res/linux/org.mixxx.Mixxx.metainfo.xml:11]: We could autogenerate it when tagging.

File: res/linux/org.mixxx.Mixxx.metainfo.xml
Comment (uklotzde) [reply res/linux/org.mixxx.Mixxx.metainfo.xml:10]: Then flathub is doing it wrong, forcing us to add a dummy value.

Comment (uklotzde) [reply res/linux/org.mixxx.Mixxx.metainfo.xml:10]: Please add a comment that explains why this element is needed, although it is optional.

Comment (Be-ing) [reply res/linux/org.mixxx.Mixxx.metainfo.xml:10]: It is not wrong. GNOME requires explicitly specifying that that there is no content rating to declare.

File: res/linux/org.mixxx.Mixxx.metainfo.xml
Comment (Be-ing) [reply res/linux/org.mixxx.Mixxx.metainfo.xml:11]: You're right, this is incorrect.

File: res/linux/org.mixxx.Mixxx.metainfo.xml
Comment (uklotzde) [inline res/linux/org.mixxx.Mixxx.metainfo.xml:13]: Please add an entry to the release checklist for updating this file.

Including the actual release date causes more work, because we have to add it right before tagging the release.

Comment (Holzhaus) [reply res/linux/org.mixxx.Mixxx.metainfo.xml:13]: Is this `<releases>` tag actually required? We should try to avoid it.

Let's say we want to release 2.3.1. This means we need to publish the release announcement before actually tagging the release here, otherwise we can't add the link to this file.

Comment (Be-ing) [reply res/linux/org.mixxx.Mixxx.metainfo.xml:13]: Yes, `releases` is a requirement. The date is not a requirement though. I think we should copy the changelog here rather than use the URL to the announcement. That way we can update this file before tagging.

Comment (uklotzde) [reply res/linux/org.mixxx.Mixxx.metainfo.xml:13]: Then let's omit the date.

The `<releases>` tag is not required, at least not when packaging for a regular Linux distribution. Version information is usually defined sparately by the packaging and not inline. If ever possible I would like to omit it entirely.

Comment (Holzhaus) [reply res/linux/org.mixxx.Mixxx.metainfo.xml:13]: Maybe this is desired: https://github.com/Be-ing/mixxx/pull/57

**Review by uklotzde — COMMENTED** [review]
Merging the generated org.mixxx.Mixxx.metainfo.xml might become difficult. Changes in the release history need to be ignored and re-generated in the target branch. Did you consider this?

**Comment by Holzhaus** [comment]
I think it should be straightforward:

1. Merge conflict in the metainfo.xml file
2. Instead of running `git mergetool`, just add the metainfo file *including the conflict markers* to the staging  area
3. Now run `pre-commit run metainfo` (all changes are staged)
4. The hook regenerates the metainfo.xml
5. `git add` the changes and commit

**Comment by uklotzde** [comment]
How about adding a comment before the `<releases>` start tag that explains that the contents are generated by `pre-commit run metainfo`? This is not obvious and not documented.

File: res/linux/org.mixxx.Mixxx.metainfo.xml
Comment (uklotzde) [inline res/linux/org.mixxx.Mixxx.metainfo.xml:2]: Please remove this old comment.

**Comment by Be-ing** [comment]
> How about adding a comment before the <releases> start tag that explains that the contents are generated by pre-commit run metainfo? This is not obvious and not documented.

Good idea, done.

**Comment by Be-ing** [comment]
Something weird is going on with the new pre-commit hook. It passed when I made the last commit, but now GitHub Actions is not passing. I downloaded and applied the patch from GitHub Actions, then tried to commit but when the hook ran again locally it canceled out the patch??

**Comment by Holzhaus** [comment]
Did you rebase or something. The changes look legit, not some miniscule formatting issues. So I suppose there is a difference in the changelog versions.

**Comment by uklotzde** [comment]
Locally the metainfo hook for this branch passes. We cannot merge before the GitHub issues are fixed.

**Comment by Be-ing** [comment]
I removed the commit with the manual changes and reran the script. Now pre-commit passes.

**Comment by Be-ing** [comment]
Please don't let this linger any longer. I am tired of having rebase and rerun the script.

File: tools/update_metainfo.py
Comment (Swiftb0y) [inline tools/update_metainfo.py:15]: I struggle to understand the `[^\)]+` part.

Comment (Holzhaus) [reply tools/update_metainfo.py:15]: I recommend regex101.com: 
Screenshot from 2021-09-08 17-08-07

It's supposed to match any char except `)` one or more times (for the milestone URL):
Screenshot from 2021-09-08 17-13-14

**Review by uklotzde — APPROVED** [review]
Ok for me.

**Comment by Be-ing** [comment]
Merge?

**Pull request merged**
